include/nanobind/stl/chrono.h

```diff
@@ -0,0 +1,439 @@
+/*
+    nanobind/stl/chrono.h: Transparent conversion between std::chrono and python's datetime
+
+    Copyright (c) 2016 Trent Houliston <trent@houliston.me> and
+                       Wenzel Jakob <wenzel.jakob@epfl.ch>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+#pragma once
+
+#include <nanobind/nanobind.h>
+
+#define __STDC_WANT_LIB_EXT1__ 1 // for localtime_s
+#include <time.h>
+
+#include <chrono>
+#include <cmath>
+#include <ctime>
+#include <limits>
+#include <type_traits>
+
+#ifndef Py_LIMITED_API
+#include <datetime.h>
+#endif
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+NAMESPACE_BEGIN(detail)
+
+#ifdef Py_LIMITED_API
+
+// <datetime.h> doesn't export any symbols under the limited API, so we'll
+// do it the hard way.
+
+struct datetime_types_t {
+    // Types defined by the datetime module
+    handle datetime;
+    handle time;
+    handle date;
+    handle timedelta;
+
+    explicit operator bool() const { return datetime.ptr() != nullptr; }
+
+    bool do_import() {
+        auto mod = module_::import_("datetime");
+        object datetime_o, time_o, date_o, timedelta_o;
+        try {
```

Comment:
I changed the unpack functions to raise an exception, which makes the false-with-error vs false-without-error return path distinction less confusing IMO. Pack still uses the Python error indicator so that 'return pack_thing(...);' at the end of a type caster works nicely.